src/Microsoft.AspNetCore.Hosting/WebHostBuilder.cs

```diff
@@ -201,6 +201,8 @@ private IServiceCollection BuildHostingServices()
             // Conjure up a RequestServices
             services.AddTransient<IStartupFilter, AutoRequestServicesStartupFilter>();
 
+            services.AddTransient<IServiceProviderFactory, DefaultServiceProviderFactory>();
```

Comment:
Since you've opted for a `ConfigureServices` call for `Use[Container]` extensions, won't this rely on ordering? Is that something that's part of the spec?

I think it would be better to move this below the `_configureServicesDelegates` code and do a `TryAddTransient`